scripts/validate_string_concatenation.py

```diff
@@ -102,28 +106,157 @@ def strings_to_concatenate(source_path: str) -> Generator[Tuple[str, int], None,
+    )
+
+    with open(source_path, "r") as file_name:
+        tokens: List = list(tokenize.generate_tokens(file_name.readline))
+
+    for first_token, second_token, third_token in zip(tokens, tokens[1:], tokens[2:]):
+        if (
+            first_token[0] == third_token[0] == token.STRING
+            and second_token[0] == token.NL
+        ):
+            # Means we are in a block of concated string
+
+            # Striping the quotes
+            first_string = first_token[1][1:-1]
+            second_string = third_token[1][1:-1]
+
+            if (not first_string.endswith(" ")) and (second_string.startswith(" ")):
+                yield source_path, third_token[2][0], MSG
+
+
+def bare_pytest_raises(source_path: str) -> Generator[Tuple[str, int, str], None, None]:
+    """
+    Test Case for bare pytest raises.
+
+    For example:
+
+    >>> with pytest.raise(ValueError):
+    ...     # Some code that raises ValueError
+
+    Instead of:
+
+    >>> with pytest.raise(ValueError, match="foo"):
+    ...     # Some code that raises ValueError
+
+    Parameters
+    ----------
+    source_path : str
+        File path pointing to a single file.
+
+    Yields
+    ------
+    source_path : str
+        Source file path.
+    line_number : int
+        Line number of unconcatenated string.
+    MSG : str
+        Explenation of the error.
+
+    Notes
+    -----
+    GH #23922
+    """
+    MSG: str = "Bare pytests raise have been found."
+
+    with open(source_path, "r") as file_name:
+        tokens: List = list(tokenize.generate_tokens(file_name.readline))
+
+    for counter, current_token in enumerate(tokens, start=1):
+        if current_token[0] == token.NAME and current_token[1] == "raises":
+            for next_token in tokens[counter:]:
+                if next_token[0] == token.NAME and next_token[1] == "match":
+                    break
+                if next_token[0] == token.NEWLINE:
+                    yield source_path, current_token[2][0], MSG
+                    break
 
 
 if __name__ == "__main__":
-    parser = argparse.ArgumentParser(description="Validate concatenated strings")
+    FUNCTIONS_MAP = {
+        "strings_to_concatenate": strings_to_concatenate,
+        "strings_with_wrong_placed_space": strings_with_wrong_placed_space,
+        "bare_pytest_raises": bare_pytest_raises,
+    }
+
+    parser = argparse.ArgumentParser(description="Unwanted patterns checker.")
 
     parser.add_argument(
         "path", nargs="?", default=".", help="Source path of file/directory to check."
     )
     parser.add_argument(
         "--format",
         "-f",
-        default="{source_path}:{line_number}:{msg}",
-        help="Output format of the unconcatenated strings.",
+        default="{source_path}:{line_number}:{msg}.",
+        help="Output format of the error message.",
+    )
+    parser.add_argument(
+        "--id", "-i", choices=FUNCTIONS_MAP.keys(), help="Test case to check."
```

Comment:
I don't think `id` is very clear, `validation-type` or something similar is more descriptive. I'd also avoid calling them "test cases", since that is confusing with unit testing terminology.